src/nvim/json/fpconv.c

```diff
@@ -0,0 +1,211 @@
+/* fpconv - Floating point conversion routines
```

Comment:
Similar to [src/xdiff/](https://github.com/neovim/neovim/issues/11239) and [src/mpack/](https://github.com/neovim/neovim/issues/15566),  it would make sense for this subsystem to live in `src/json/` . The rationale is that this signals to contributors that we don't own these modules. We may add a few tweaks such as `mpack.NIL` handling, but otherwise try to keep this code sync'd with upstream. And try to avoid dependencies on `src/nvim/` (if possible).

see also
- https://github.com/neovim/neovim/issues/11239 
- https://github.com/neovim/neovim/pull/15566#discussion_r703069659

Reply:
Added benefit: make it easier/clearer to apply Neovim style guide (uncrustify) only `src/nvim`, not external subsystems.

Reply:
done